crates/wasmer-provider/src/pipes.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    convert::TryFrom,
+    fmt::Debug,
+    io::{self, Read, Seek, Write},
+};
+
+use anyhow::Error;
+use serde::{Deserialize, Serialize};
+use tokio::fs::File;
+use wasmer_wasi::{VirtualFile, WasiFsError};
+
+/// For piping stdio. Stores all output / input in a byte-vector.
+pub struct FilePipe {
+    file: std::fs::File,
+}
+
+impl Debug for FilePipe {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FilePipe")
+            .field("file", &self.file)
+            .finish()
+    }
+}
+
+impl Serialize for FilePipe {
+    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        unimplemented!()
+    }
+}
+
+impl<'de> Deserialize<'de> for FilePipe {
+    fn deserialize<D>(_deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        unimplemented!()
+    }
+}
+
+impl TryFrom<File> for FilePipe {
+    type Error = Error;
+
+    fn try_from(value: File) -> Result<Self, Self::Error> {
+        Ok(Self {
+            file: value
+                .try_into_std()
```

Comment:
So this will immediately attempt to convert a File to a StdFile without allowing async write operations to complete IIRC. Is there a technical reason why we can't implement From instead of TryFrom to allow a cleaner migration?

https://docs.rs/tokio/1.12.0/tokio/fs/struct.File.html#method.try_into_std